### PR TITLE
Feature/136 resource gantt chart frontend integration

### PR DESCRIPTION
**Pull request by loststoner**
## Feature: Resource Gantt-chart frontend integration #136 

Implemented Resource Gantt-chart with Tab Panel

Screenshot from 2024-09-24 11-14-13

### CODE REVIEW

File: src/templates/dashboard/job_task_gantt.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        },
+        vUseSingleCell: 10000, // Set the threshold cell per table row (Helps performance for large data.
+        vFormatArr: ['Day', 'Week', 'Month', 'Quarter'], // Even with setUseSingleCell using Hour format on such a large chart can cause issues in some browsers,
+
+        vMinGpLen: 20,
+    });
+
+
+    // to get data from the api and display it in the gantt chart
+    axios.get('http://localhost:8000/api/job/gantt')
```
Comment (riddick-oe) [inline src/templates/dashboard/job_task_gantt.html:34]: This URL wouldn't be accessible in production. We need to get this from an external source, like an environment variable.

File: src/templates/dashboard/resource_gantt.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        vShowTaskInfoComp: 0,
+        vUseSingleCell: 10000, // Set the threshold cell per table row (Helps performance for large data.
+        vFormatArr: ['Day', 'Week', 'Month', 'Quarter'], // Even with setUseSingleCell using Hour format on such a large chart can cause issues in some browsers,
+
+        vMinGpLen: 20,
+    });
+
+
+    // to get data from the api and display it in the gantt chart
+    axios.get('http://localhost:8000/api/resource/gantt')
```
Comment (riddick-oe) [inline src/templates/dashboard/resource_gantt.html:28]: Same comment: https://github.com/factryflow/factryflow/pull/167/files#r1772677051

File: src/templates/dashboard/job_task_gantt.html (hunk 1 above)
Comment (loststoner) [reply src/templates/dashboard/job_task_gantt.html]: fixed

File: src/templates/dashboard/resource_gantt.html (hunk 2 above)
Comment (loststoner) [reply src/templates/dashboard/resource_gantt.html]: fixed

**Review by riddick-oe — APPROVED** [review]

**Pull request merged**